tokio-postgres/src/row.rs

```diff
@@ -100,6 +100,63 @@ pub struct Row {
     ranges: Vec<Option<Range<usize>>>,
 }
 
+/// A macro to map pg types to rust types, for debug display.
+macro_rules! debug_row_type {
+    ($this:expr; $map:expr; $idx:expr; $name:expr; $type:expr; $($pg_ty:tt => $ty:ty),*) => {
+        match $type {
+            $(
+                &Type::$pg_ty => match <$ty as FromSql>::from_sql_nullable(
+                    &Type::$pg_ty,
+                    $this.0.col_buffer($idx),
+                ) {
+                    Ok(val) => $map.entry(&$name, &val),
+                    Err(_) => $map.entry(&$name, &"<error>"),
+                },
+            )*
+            _ => $map.entry(&$name, &"<opaque type>"),
+        }
+    }
+}
+
+impl fmt::Debug for Row {
```

Comment:
This is a lot of code for a debug impl.

Reply:
Yes it's a little bit complicated. The reason is that it inspects the postgres type of the column value, and then chooses an appropriate debuggable rust type to convert into, if possible. It gives output like the following:

```text
Row {
    columns: [
        Column {
            name: "id",
            type: Int4,
        },
        Column {
            name: "created_at",
            type: Timestamptz,
        },
        Column {
            name: "message",
            type: Varchar,
        },
        Column {
            name: "email",
            type: Varchar,
        },
        Column {
            name: "name",
            type: Varchar,
        },
    ],
    data: {
        "id": 1,
        "created_at": SystemTime {
            tv_sec: 1577464626,
            tv_nsec: 0,
        },
        "message": "test",
        "email": "test",
        "name": "test",
    },
}
```

Reply:
I only did a few of the types where I knew what rust type to use, the rest will return `<opaque type>`.

Reply:
I'd rather not try to downcast the data for some subset of types. Just displaying the column information seems good enough IMO.

Reply:
I'll make the change you request. Still, seems a shame to not make use of the extra data.